addons/mass_mailing/models/mailing.py

```diff
@@ -1341,6 +1341,21 @@ def _image_to_url(b64image: bytes):
 
                 node.attrib['src'] = _image_to_url(image)
                 modified = True
+        # handle the images in mso comments
+        for comment in root.xpath('//comment()'):
+            is_mso = mso_re.match(comment.text)
+            if is_mso:
+                matches = re.findall(r'<img[^>]*src="(data:image/[A-Za-z]+;base64,[^"]*)"', comment.text)
+                if matches:
+                    modified = True
+                    for match in matches:
+                        image = re.sub(r'data:image/[A-Za-z]+;base64,', '', match).encode()  # base64 image as bytes
+                        comment.text = comment.text.replace(match, _image_to_url(image))
```

Comment:
Isn't there a way to swap the order of things ? So that we create mso comments based on images that already have the correct values, therefore we don't have to do the same adaptations twice.

Note that here you call one of the images adaptation methods but not the other (I think it was crop ?), is it intended ?

Reply:
> Isn't there a way to swap the order of things ? So that we create mso comments based on images that already have the correct values, therefore we don't have to do the same adaptations twice.

Well the mso comments are created in the conversion process in js, and this is done in the saving process in py. So I believe the answer is yes but not without requiring a big change in the whole flow (refactoring this so we can call `_image_to_url` from a route so it can be called via rpc at the beginning of `convert_inline.js` (slowing it down too).

> Note that here you call one of the images adaptation methods but not the other (I think it was crop ?), is it intended ?

I'm not _calling_ one of the images adaptation methods. This is one of the images adaptations methods, which I'm correcting so it works with mso (the crop one is already only for mso anyway).